test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java

```diff
@@ -275,8 +283,37 @@ public void wipeCluster() throws Exception {
         }
 
         // wipe index templates
+        String deleteTemplates = "*";
+        if (excludeTemplates().size() > 0) {
+            try {
+                // We need to list all existing templates first
+                RestResponse response = adminExecutionContext.callApi("indices.get_template", Collections.emptyMap(),
+                        Collections.emptyList(), Collections.emptyMap());
+                try (XContentParser parser = XContentType.JSON.xContent().createParser(response.getBodyAsString())) {
+                    Set<String> deletions = new HashSet<>();
+
+                    String[] exclusions = excludeIndices().toArray(new String[]{});
+                    for (String template : parser.map().keySet()) {
+                        if (Regex.simpleMatch(exclusions, template) == false) {
+                            // If the template name does not match an exclusion,
+                            // we add it to the list of templates to be deleted
+                            deletions.add(template);
+                        }
+                    }
+
+                    if (deletions.isEmpty() == false) {
```

Comment:
but a list of exclusions was specified... is the idea here to prevent users from excluding all templates?